source/core/src/main/com/csse3200/game/screens/AnimalSelectionScreen.java

```diff
@@ -3,74 +3,96 @@
 import com.badlogic.gdx.Gdx;
 import com.badlogic.gdx.ScreenAdapter;
 import com.badlogic.gdx.graphics.GL20;
+import com.badlogic.gdx.scenes.scene2d.InputEvent;
 import com.badlogic.gdx.scenes.scene2d.Stage;
 import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
 import com.csse3200.game.GdxGame;
-import com.csse3200.game.components.animal.AnimalSelectionDisplay;
 import com.csse3200.game.components.animal.AnimalSelectionActions;
+import com.csse3200.game.components.animal.AnimalSelectionDisplay;
 import com.csse3200.game.ui.PopUpDialogBox.PopUpHelper;
 
-public class AnimalSelectionScreen extends ScreenAdapter {
-    private Stage stage;
-    private AnimalSelectionDisplay display;
-    private AnimalSelectionActions actions;
+public abstract class AnimalSelectionScreen extends ScreenAdapter {
+    protected Stage stage;
```

Comment:
This should probably set up the stage and resources using the service locator - same as the other screens.